apps/site/lib/predicted_schedule.ex

```diff
@@ -133,9 +134,11 @@ defmodule PredictedSchedule do
     {{trip_id, stop_id, stop_sequence}, ps}
   end
 
-  defp group_transform(%{trip: trip, stop: stop} = ps)
-       when not is_nil(trip) and not is_nil(stop) do
-    {{ps.trip.id, ps.stop.id, ps.stop_sequence}, ps}
+  defp group_transform(%{trip: %Trip{id: trip_id}, stop: stop} = ps) do
```

Comment:
is this a possibility for both `Schedule` and `Prediction`? or can this be narrowed down to just one type?

Reply:
I've only seen witnessed it on Schedule but theoretically it's possible on both. 🤔  Noting how we treat Predictions in the clause above it makes me wonder what should be done if a Prediction had a valid trip and nil stop. Going to look at this a bit

Reply:
Well, I was about to push an update that was more specific about handling `Schedule` vs `Prediction` but then I realized the overall logic was unchanged! 😝

In both cases, for this particular function, we want to return the same result.